standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -546,9 +548,17 @@ public static void updateTableStatsForCreateTable(Warehouse wh, Database db, Tab
       }
     }
 
-    if (MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.STATS_AUTO_GATHER)) {
+    boolean isIcebergTable = Optional.ofNullable(tbl.getParameters())
+            .orElse(Collections.emptyMap())
+            .getOrDefault(TABLE_TYPE_PROP, "undefined")
+            .equals(ICEBERG_TABLE_TYPE_VALUE);
```

Comment:
Does ``statsSetup.isIcebergTable`` doesn't tell you it is a ``iceberg`` table?

Reply:
Agree with @ayushtkn  I think this is not necessary call here.

Reply:
Done

Reply:
Hi @ayushtkn, @aturoczy,
I reverted to using previous version of checking if the table is iceberg using table type property instead of using statsSetup.isIcebergTable, because I found it didn't work for Iceberg table created as select. 
statsSetup is retrieved from table's dictionary, but when creating iceberg table as select, the table has no dictionary and this condition is false, so it is not reliable way to check if table is iceberg for CTAS table:
https://github.com/apache/hive/blob/e982ed1105062648b395c320425ea6683794f4d6/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java#L522